android_app/app/src/main/java/com/health/openscale/core/bluetooth/BluetoothOKOK2.java

```diff
@@ -0,0 +1,189 @@
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU General Public License for more details.
+ *
+ *    You should have received a copy of the GNU General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */
+package com.health.openscale.core.bluetooth;
+
+import android.bluetooth.le.ScanFilter;
+import android.bluetooth.le.ScanResult;
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+import android.util.SparseArray;
+
+import com.health.openscale.core.datatypes.ScaleMeasurement;
+import com.health.openscale.core.utils.Converters;
+import com.welie.blessed.BluetoothCentralManager;
+import com.welie.blessed.BluetoothCentralManagerCallback;
+import com.welie.blessed.BluetoothPeripheral;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import timber.log.Timber;
+
+import static com.health.openscale.core.utils.Converters.WeightUnit.LB;
+import static com.health.openscale.core.utils.Converters.WeightUnit.ST;
+
+public class BluetoothOKOK2 extends BluetoothCommunication {
+    private static final int IDX_WEIGHT_MSB = 0;
+    private static final int IDX_WEIGHT_LSB = 1;
+    private static final int IDX_IMPEDANCE_MSB = 2;
+    private static final int IDX_IMPEDANCE_LSB = 3;
+    private static final int IDX_PRODUCTID_MSB = 4;
+    private static final int IDX_PRODUCTID_LSB = 5;
+    private static final int IDX_ATTRIB = 6;
+    private static final int IDX_MAC_1 = 7;
+    private static final int IDX_MAC_2 = 8;
+    private static final int IDX_MAC_3 = 9;
+    private static final int IDX_MAC_4 = 10;
+    private static final int IDX_MAC_5 = 11;
+    private static final int IDX_MAC_6 = 12;
+
+    private static final int UNIT_KG = 0;
+    private static final int UNIT_LB = 2;
+    private static final int UNIT_STLB = 3;
+
+    private BluetoothCentralManager central;
+    private String mMacAddress;
+    private float mLastWeight = 0f;
+
+    public BluetoothOKOK2(Context context) {
+        super(context);
+        central = new BluetoothCentralManager(context, btCallback, new Handler(Looper.getMainLooper()));
+    }
+
+    private final BluetoothCentralManagerCallback btCallback = new BluetoothCentralManagerCallback() {
+        @Override
+        public void onDiscoveredPeripheral(@NotNull BluetoothPeripheral peripheral, @NotNull ScanResult scanResult) {
+            SparseArray<byte[]> manufacturerSpecificData = scanResult.getScanRecord().getManufacturerSpecificData();
+            int vendorIndex = -1;
+            for (int i = 0; i < manufacturerSpecificData.size(); i++) {
+                int vendorId = manufacturerSpecificData.keyAt(i);
+                if ((vendorId & 0xff) == 0xc0) { // 0x00c0-->0xffc0
```

Comment:
Is the 0xc0 the identifier for the scale?

Reply:
The ID is 0x??c0 (0x00c0, 0x01c0, 0x02c0.... ) where the MSB is used to notify the progress, it starts from 0 and increases until the measurement is stable